apps/nav-veileders-flate/src/api/data/forslag.ts

```diff
@@ -0,0 +1,32 @@
+import { z } from 'zod'
+
+export enum ForslagEndringType {
+  ForlengDeltakelse = 'ForlengDeltakelse',
+  EndreStartdato = 'EndreStartdato'
+}
+
+export const forlengDeltakelseForslagSchema = z.object({
+  type: z.literal(ForslagEndringType.ForlengDeltakelse),
+  sluttdato: z.string()
+})
+
+export const endreStartdatoForslagSchema = z.object({
+  type: z.literal(ForslagEndringType.EndreStartdato),
+  startdato: z.string()
+})
+
+export const forslagEndringSchema = forlengDeltakelseForslagSchema.or(
```

Comment:
Så ut som at jeg måtte ha minst to verdier her, så da la jeg bare til en annen type 😅

Reply:
Sånn har jeg gjort schema for `AktivtForslag` med endringer og status, da skal det bare være å fylle inn listen på `endringSchema` etter hvert som det legges til flere endringer, og samme for statuser.
```typescript
import { z } from 'zod'

import { dateSchema } from '../utils'

export enum ForslagStatusType {
	VenterPaSvar = 'VenterPaSvar',
}

export enum ForslagEndringType {
	ForlengDeltakelse = 'ForlengDeltakelse',
}


const forlengDeltakelseSchema = z.object({
	type: z.literal(ForslagEndringType.ForlengDeltakelse),
	sluttdato: dateSchema,
})

const endringSchema = z.discriminatedUnion('type', [
	forlengDeltakelseSchema,
])

const venterPaSvarSchema = z.object({
	type: z.literal(ForslagStatusType.VenterPaSvar),
})

const statusSchema = z.discriminatedUnion('type', [
	venterPaSvarSchema,
])

export const aktivtForslagSchema = z.object({
	id: z.string().uuid(),
	opprettet: dateSchema,
	begrunnelse: z.string().nullable(),
	endring: endringSchema,
	status: statusSchema.default({ type: ForslagStatusType.VenterPaSvar }),
})

export type AktivtForslag = z.infer<typeof aktivtForslagSchema>
export type ForslagEndring = z.infer<typeof endringSchema>
```

Reply:
Aha! Og det var lurt å ta med status. Jeg tenkte at man aldri vil få noe annet enn "VenterPaSvar" her, men når man har med status kan man jo bruke samme formatet for endringsloggen :)